tests/epics/test_signals.py

```diff
@@ -614,3 +619,76 @@ async def test_str_returns_enum(ioc: IOC):
     assert val == MyEnum.b
     assert val == "Bbb"
     assert val is not MyEnum.b
+
+
+async def test_signal_returns_units_and_precision(ioc: IOC):
+    await ioc.make_backend(float, "float")
+    pv_name = f"{ioc.protocol}://{PV_PREFIX}:{ioc.protocol}:float"
+
+    sig = epics_signal_rw(float, pv_name)
+    await sig.connect()
+    datakey = await sig.describe()[""]
+    assert datakey["units"] == "mm"
+    assert datakey["precision"] == 1
+
+
+async def test_signal_not_return_none_units_and_precision(ioc: IOC):
+    await ioc.make_backend(str, "str")
+    pv_name = f"{ioc.protocol}://{PV_PREFIX}:{ioc.protocol}:str"
+
+    sig = epics_signal_rw(str, pv_name)
+    await sig.connect()
+    datakey = await sig.describe()[""]
+    assert not hasattr(datakey, "units")
+    assert not hasattr(datakey, "precision")
+
+
+async def test_signal_returns_limits(ioc: IOC):
+    await ioc.make_backend(int, "int")
+    pv_name = f"{ioc.protocol}://{PV_PREFIX}:{ioc.protocol}:int"
+
+    expected_limits = Limits(
+        # LOW, HIGH
+        alarm=LimitPair(low=2, high=98),
+        # DRVL, DRVH
+        control=LimitPair(low=10, high=90),
+        # LOPR, HOPR
+        display=LimitPair(low=0, high=100),
+        # LOLO, HIHI
+        warning=LimitPair(low=5, high=96),
+    )
+
+    sig = epics_signal_rw(int, pv_name)
+    await sig.connect()
+    limits = (await sig.describe())[""]["limits"]
+    assert limits == expected_limits
+
+
+async def test_signal_returns_partial_limits(ioc: IOC):
+    await ioc.make_backend(int, "pint")
+    pv_name = f"{ioc.protocol}://{PV_PREFIX}:{ioc.protocol}:pint"
+
+    expected_limits = Limits(
+        # LOW, HIGH, not set
+        alarm=LimitPair(),
+        # LOLO, HIHI
+        warning=LimitPair(low=5, high=96),
+        # DRVL, DRVH, not set
+        control=LimitPair(),
+        # LOPR, HOPR
+        display=LimitPair(low=0, high=100),
+    )
```

Comment:
@coretl are these mappings correct?

Reply:
For PVA, with LOW, HIGH, LOLO, HIHI set: `alarm.low=alarm.high=warning.low=warning.high=nan`
For VA, with LOW, HIGH, LOLO, HIHI set: `alarm.low=alarm.high=warning.low=warning.high=0`